src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -368,24 +369,30 @@ public void sendActionListenerOnFailureRequest(TransportService transportService
     }
 
     /**
-     * Requests the environment setting values from OpenSearch for the corresponding component settings. The result will be handled by a {@link EnvironmentSettingsResponseHandler}.
+     * Requests the environment settings from OpenSearch. The result will be handled by a {@link EnvironmentSettingsResponseHandler}.
      *
-     * @param componentSettings The component setting that correspond to the values provided by the environment settings
      * @param transportService  The TransportService defining the connection to OpenSearch.
+     * @return A Setting object from the OpenSearch Node environment
      */
-    public void sendEnvironmentSettingsRequest(TransportService transportService, List<Setting<?>> componentSettings) {
+    public Settings sendEnvironmentSettingsRequest(TransportService transportService) {
+        final CountDownLatch inProgressLatch = new CountDownLatch(1);
         logger.info("Sending Environment Settings request to OpenSearch");
-        EnvironmentSettingsResponseHandler environmentSettingsResponseHandler = new EnvironmentSettingsResponseHandler();
+        EnvironmentSettingsResponseHandler environmentSettingsResponseHandler = new EnvironmentSettingsResponseHandler(inProgressLatch);
         try {
             transportService.sendRequest(
                 opensearchNode,
                 ExtensionsOrchestrator.REQUEST_EXTENSION_ENVIRONMENT_SETTINGS,
-                new EnvironmentSettingsRequest(componentSettings),
+                new ExtensionRequest(ExtensionsOrchestrator.RequestType.REQUEST_EXTENSION_ENVIRONMENT_SETTINGS),
                 environmentSettingsResponseHandler
             );
+            // Wait on environment settings response
+            inProgressLatch.await(100, TimeUnit.SECONDS);
         } catch (Exception e) {
             logger.info("Failed to send Environment Settings request to OpenSearch", e);
         }
```

Comment:
Since you're adding an `await()` you should explicitly handle the `InterruptedException`.